src/main/java/menion/android/whereyougo/gui/activity/MainActivity.java

```diff
@@ -399,6 +691,43 @@ public boolean onOptionsItemSelected(MenuItem item) {
         }
     }
 
+    @Override
+    public void onDestroy() {
+        if (finish) {
+            // stop debug if any forgotten
```

Comment:
The call sequence is opposite - first the part in (old) MainActivity.onDestroy, there is the call to super.onDestroy()=CustomMainActivity.onDestroy().

Reply:
Addressed it in the force push.

Reply:
Please check the original code, the "then" part is now necessary.

Reply:
You need the part
~~~
        } else {
            super.onDestroy();
~~~
see the original implementation.

Reply:
Added